src/structured-data/handler.js

```diff
@@ -149,33 +191,176 @@ export async function opportunityAndSuggestions(auditUrl, auditData, context) {
 }
 
 export async function structuredDataHandler(baseURL, context, site) {
-  const { log } = context;
+  const { log, dataAccess } = context;
   const startTime = process.hrtime();
 
   const siteId = site.getId();
 
-  const structuredDataURLs = await site.getConfig().getIncludedURLs(auditType);
-  if (structuredDataURLs && isArray(structuredDataURLs) && structuredDataURLs.length === 0) {
-    log.error(`No product detail pages found for site ID: ${siteId}`);
-    throw new Error(`No product detail pages found for site: ${baseURL}`);
+  try {
+    const topPages = await getTopPagesForSiteId(dataAccess, siteId, context, log);
+    if (topPages.length === 0) {
+      log.error(`No top pages for site ID ${siteId} found. Ensure that top pages were imported.`);
+      throw new Error(`No top pages for site ID ${siteId} found.`);
+    }
+
+    const auditResult = await processStructuredData(baseURL, context, topPages);
+
+    const endTime = process.hrtime(startTime);
+    const elapsedSeconds = endTime[0] + endTime[1] / 1e9;
+    const formattedElapsed = elapsedSeconds.toFixed(2);
+
+    log.info(`Structured data audit completed in ${formattedElapsed} seconds for ${baseURL}`);
+
+    return {
+      fullAuditRef: baseURL,
+      auditResult,
+    };
+  } catch (e) {
+    return {
+      fullAuditRef: baseURL,
+      auditResult: {
+        error: e.message,
+        success: false,
+      },
+    };
   }
+}
 
-  const auditResult = await processStructuredData(baseURL, context, structuredDataURLs);
+export async function generateSuggestionsData(auditUrl, auditData, context, site) {
+  const { dataAccess, log } = context;
+  const { Configuration } = dataAccess;
 
-  const endTime = process.hrtime(startTime);
-  const elapsedSeconds = endTime[0] + endTime[1] / 1e9;
-  const formattedElapsed = elapsedSeconds.toFixed(2);
+  // Check if audit was successful
+  if (auditData.auditResult.success === false) {
+    log.info('Audit failed, skipping suggestions data generation');
+    return { ...auditData };
+  }
 
-  log.info(`Structured data audit completed in ${formattedElapsed} seconds for ${baseURL}`);
+  // Check if auto suggest was enabled
+  const configuration = await Configuration.findLatest();
+  if (!configuration.isHandlerEnabledForSite('structured-data-auto-suggest', site)) {
```

Comment:
possibly should go into the `Audit` model for centralized referencing: https://github.com/adobe/spacecat-shared/blob/main/packages/spacecat-shared-data-access/src/models/audit/audit.model.js#L26

Reply:
👍 Added via https://github.com/adobe/spacecat-shared/pull/628. Will update PR once the other one is merged.